pkg/kube/controllers/extendedstatefulset/pod_mutator.go

```diff
@@ -141,30 +150,45 @@ func (m *PodMutator) mutatePodsFn(ctx context.Context, pod *corev1.Pod) error {
 
 	podOrdinal := names.OrdinalFromPodName(pod.GetName())
 	if podOrdinal != -1 {
-		podLabels[essv1.LabelPodOrdinal] = strconv.Itoa(podOrdinal)
+		podLabels[essv1a1.LabelPodOrdinal] = strconv.Itoa(podOrdinal)
 		pod.SetLabels(podLabels)
 	}
-
 	return nil
 }
 
-// changeVolumePods will append volume and change volumeMount name
-func changeVolumePods(desiredPVCName string, pod *corev1.Pod, desiredVersionInt int, currentVersionInt int, volumeClaimTemplate *corev1.PersistentVolumeClaim) {
+// addVolumeSpec adds volume spec to the pod container volumes spec
+func (m *PodMutator) addVolumeSpec(pod *corev1.Pod, volumeClaimTemplatesMap map[string]corev1.PersistentVolumeClaim, volumeMap map[string]corev1.Volume, statefulSet *v1beta2.StatefulSet) {
 
-	// generate desired vct name
-	desiredVersion := fmt.Sprintf("%s%d", "v", desiredVersionInt)
-	desiredVCTName := replaceVersionInName(volumeClaimTemplate.Name, desiredVersion, 1)
+	for _, container := range pod.Spec.Containers {
+		for _, volumeMount := range container.VolumeMounts {
 
-	appendVolumetoPod(pod, volumeClaimTemplate, desiredVCTName, desiredPVCName)
-	removeUnusedVolumes(pod, desiredVCTName, volumeClaimTemplate.Name)
-}
+			_, foundVolumeClaimTemplate := volumeClaimTemplatesMap[volumeMount.Name]
+			if foundVolumeClaimTemplate {
+				podOrdinal := pod.Name[len(pod.Name)-1:]
```

Comment:
Get `podOrdinal` from `names.OrdinalFromPodName(pod.GetName())` is better

Reply:
Resolved.